lib/spack/llnl/util/filesystem.py

```diff
@@ -198,15 +198,42 @@ def getuid():
         return os.getuid()
 
 
+def _win_rename(src, dst):
+    tmp_dir = ""
+    # os.replace will still fail if on Windows (but not POSIX) if the dst
+    # is a symlink to a directory (all other cases have parity Windows <-> Posix)
+    @contextmanager
+    def safe_rename(src, dst):
+        if os.path.isdir(os.path.relapath(dst)):
+            # If dst exists and is a symlink to a directory
+            # we need to remove dst and rename/replace
+            # if dst == src (i.e. they're the same symlink)
+            # removing dst directly will delete src, copy src to a temp directory
+            # and perform rename from there.
+            tmp_dir = tempfile.mkdtemp()
+            tmp_copy = os.path.join(tmp_dir, os.path.basename(src))
+            shutil.copyfile(src, tmp_copy) if os.path.isfile(src) else copy_tree(src, tmp_copy)
```

Comment:
(comments about not recovering supersede this, but ...)

* We own the tmp dir, so we should `mv` the file there, vs. copy it (it could be expensive to copy a whole dir)
* `rename` on to a dir should fail, if the dst is a symlink and we want that to succeed on Windows (to be consistent w/ Linux), I think copy would be suitable anyway (we'd be copying the link)

Reply:
I'm confused by this comment, my understanding is:
You suggest we should `mv` `src` to `tmp_dir` rather than copy, but copy would be suitable? For replacing `dst`?